examples/helloworld/greeting/greet_host.pb.go

```diff
@@ -16,40 +16,54 @@ import (
 	api "github.com/tetratelabs/wazero/api"
 	wasi_snapshot_preview1 "github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
 	sys "github.com/tetratelabs/wazero/sys"
-	io "io"
-	fs "io/fs"
 	os "os"
 )
 
 const GreeterPluginAPIVersion = 1
 
-type GreeterPluginOption struct {
-	Stdout io.Writer
-	Stderr io.Writer
-	FS     fs.FS
-}
+type GreeterPluginOption func(plugin *GreeterPlugin)
 
 type GreeterPlugin struct {
-	cache  wazero.CompilationCache
-	config wazero.ModuleConfig
+	newRuntime   func(context.Context) (wazero.Runtime, error)
+	cache        wazero.CompilationCache
```

Comment:
since cache is a part of runtime config I might simplify things by removing this for a godoc example of setting it via newRuntime. We don't have to keep compat yet, and we've broken recently anyway. sound ok @evacchi?

Reply:
yes, explicit `cache` feels redundant with `newRuntime` here

Reply:
Thanks, @evacchi . `cache` parameter deleted. Do we need to call `defer r.Close()` in this case?
Also seems to plugin `Close(ctx context.Context)` can be removed?

Reply:
Oh, my bad. We should close runtime at the end of invocation, not there.